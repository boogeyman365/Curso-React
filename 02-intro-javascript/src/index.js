const nombre = 'Oscar';
const apellido = 'Establie';

const nombreCompleto = `${nombre} ${apellido}`

console.log(nombreCompleto);