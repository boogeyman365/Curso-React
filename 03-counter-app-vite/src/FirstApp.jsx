// import { Fragment } from 'react';
import PropTypes from 'prop-types';

// export const FirstApp = () => {
//   return (
//   // Uso de Fragment para que no cree un div fantasma en el HTML
//     <Fragment>
//       <h1>Oscar</h1>
//       <p>Soy un subtitulo</p>
//     </Fragment>
//   );
// }


const getResult = (a , b) => {
  return (a + b)
};

export const FirstApp = ({ title, subTitle, name }) => {

  return (
    // Usar <> es lo mismo que usar Fragment sin tener que importarlo
    <>
      <h1 data-testid="test-title"> { title } </h1>
      <h4> { subTitle } </h4>
      <h4> { name } </h4>
      {/* <p>Soy un { getResult(1,20) }</p> */}
    </>
  );
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number,
}

FirstApp.defaultProps = {
  title: 'Title por defecto',
  subTitle: 123,
  name: 'Oscar establie'
}