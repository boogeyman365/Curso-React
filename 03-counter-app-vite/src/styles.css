html,body {
  background-color: #21232A;
  color: white;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 20px;
}

button {
  background-color: #53a576;
  border-radius: 5px;
  cursor: pointer;
  font-size: 1.3rem;
  padding: 15px;
  margin-right: 5px;
}