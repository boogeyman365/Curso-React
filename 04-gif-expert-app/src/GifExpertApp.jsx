import { useState } from "react"
import { AddCategory } from "./Components/addCategory";
import { GifGrid } from "./Components/GifGrid";

export const GifExpertApp = () => {

    const [ categories, setCategories ] = useState([ 'One Punch' ]);

    const onAddCategory = (newCategory) => {
        // console.log(newCategory);
        if (categories.includes(newCategory)) return;
        setCategories( [newCategory, ...categories] );
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory 
                onNewCategory={ onAddCategory }
            ></AddCategory>

            { 
                categories.map( ( category ) => ( 
                    <GifGrid 
                        key={category} 
                        category={category}
                    ></GifGrid> 
                )) 
            }

            {/* Gif Item */}
        </>
    )
}