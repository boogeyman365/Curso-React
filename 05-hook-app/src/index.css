body {
  padding: 20px;
}

button {
  margin-right: 10px;
}

.link {
  margin-right: 10px;
}