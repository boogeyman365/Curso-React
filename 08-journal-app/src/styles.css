

.box-shadow {
  box-shadow: 0px 5px 5px rgba(0, 0, 0, 0.2);
}

.active-note {
  background-color: rgba(8, 116, 116, 0.603);
}