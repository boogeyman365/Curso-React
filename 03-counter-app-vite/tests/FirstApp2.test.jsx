import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp', () => {
  
  const title = 'Hola Mundo';

  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<FirstApp title={title}></FirstApp>);

    expect(container).toMatchSnapshot();
  })

  test('Debe de mostrar el mensaje "Hola soy Goku"', () => {
    render(<FirstApp title={title}></FirstApp>);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Debe de mostrar el titulo en un H1', () => {
    render(<FirstApp title={title}></FirstApp>);
    expect(screen.getByRole('heading',{level: 1}).innerHTML).toContain(titleu);
  })
  
  
})
