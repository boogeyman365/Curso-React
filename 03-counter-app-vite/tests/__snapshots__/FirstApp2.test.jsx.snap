// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola Mundo
     
  </h1>
  <h4>
     
    123
     
  </h4>
  <h4>
     
    Oscar establie
     
  </h4>
</div>
`;
